File: assignments_nilesh_sir/java_assignments/Assignment_3/Hospital/Patient.java
package Hospital;

import Hospital.BedOptions;

public class Patient {
    private static int patientId;
    private String patientName;
    private BedOptions bedType;
    private int noOfDays;

    public Patient() {
        patientId++;
        bedType = BedOptions.ECONOMICAL;
    }

    public int getId() {
        return patientId;
    }

    public void setName(String name) {
        patientName = name;
    }

    public void setNoOfDays(int days) {
        noOfDays = days;
    }

    public String getName() {
        return patientName;
    }

    public int getNoOfDays() {
        return noOfDays;
    }

    public void setBed(BedOptions bed) {
        bedType = bed;
    }

    public int getPricePerDay() {
        switch (bedType) {
            case EXECUTIVE:
                return 500;
            case SPECIAL:
                return 350;
            default:
                return 200;
        }
    }

    public double getBillAmount() {
        double amount;
        amount = getNoOfDays() * getPricePerDay();
        return amount;
    }
}

File: assignments_nilesh_sir/Dotnet_assignments/Assignment_2/Patient.cs
using System;
class Patient
{
    private static int Count = 0;

    public int PatientId { get; } = ++Count;

    public string Pname { get; set; }

    public BedType Bed { get; set; }
    public int NoOfDays { get; set; }

    public int GetPricePerDay
    {
        get
        {
            switch (Bed)
            {
                case BedType.EXECUTIVE:
                    return 500;
                case BedType.SPECIAL:
                    return 350;
                default:
                    return 200;
            }
        }
    }

    public virtual double BillAmount => NoOfDays * GetPricePerDay;

    public void Display()
    {
        Console.WriteLine("Patient {0} with ID {1} has been admitted for {2} days with Bed as {3} and the bill total is: {4} ", Pname, PatientId, NoOfDays, Bed, BillAmount);
    }
}